packages/canisters/src/ic_af/src/lib.rs

```diff
@@ -67,17 +27,47 @@ fn sha256(input: &[u8]) -> [u8; 32] {
 /// verifying the signatures for data-package
 /// Sample file : ../../../../fixtures/data-package.json
 #[update]
-fn verify_data_proof(data_package : String) -> (String, String) {
-    let data_package : DataPackage = serde_json::from_str(data_package.as_str()).unwrap();
-    // todo!();
-    (String::new(), String::new())
+async fn verify_data_proof(redstone_data : String) -> (Vec<DataPackage>, String) {
+    let redstone_data : RedstoneData = serde_json::from_str(redstone_data.as_str()).unwrap();
+    let mut signer_bytes = [0u8; 20];
+    let mut result = Vec::new();
+
+    hex::decode_to_slice(redstone_data.address[2..].to_owned(), &mut signer_bytes).unwrap();
+    for data_package in redstone_data.dataPackages {
+        let data_package_bytes = hex::decode(data_package.as_str()[2..].to_owned()).unwrap();
+        let data_package = DataPackage::extract_and_verify(data_package_bytes.as_slice(), &signer_bytes);
+        result.push(data_package);
+    }
+
+    let bin_data = bincode::serialize(&result).unwrap();
+
+    let request = SignWithECDSA {
+        message_hash: sha256(bin_data.as_slice()).to_vec(),
+        derivation_path: vec![],
+        key_id: EcdsaKeyIds::TestKeyLocalDevelopment.to_key_id(),
```

Comment:
Here I believe we'll need to replace the `key_id` based on the environment that the Canister is deployed in - correct?

Reply:
Yes. For release, it must be changed to `EcdsaKeyIds::key_1` according to [https://internetcomputer.org/docs/current/developer-docs/smart-contracts/encryption/signing-messages](https://internetcomputer.org/docs/current/developer-docs/smart-contracts/encryption/signing-messages).
Should I change this?

Reply:
Make this relative to the environment that the Canister is being deployed to.
Typically is a regular program, this can be environment variables.